hedera-mirror-rest-java/src/main/java/com/hedera/mirror/restjava/jooq/DomainRecordMapper.java

```
@@ -0,0 +1,184 @@
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.Nullable;
+import org.jooq.EnumType;
+import org.jooq.JSONB;
+import org.jooq.Record;
+import org.jooq.RecordMapper;
+import org.jooq.RecordType;
+import org.jooq.postgres.extensions.types.LongRange;
+
+class DomainRecordMapper<R extends Record, E> implements RecordMapper<R, E> {
+
+    private static final Converter<String, String> FORMAT_CONVERTER =
+            CaseFormat.LOWER_CAMEL.converterTo(CaseFormat.LOWER_UNDERSCORE);
+    private static final MethodHandles.Lookup LOOKUP = MethodHandles.lookup();
+    private static final ObjectMapper OBJECT_MAPPER = ObjectToStringSerializer.OBJECT_MAPPER;
+
+    private final MethodHandle defaultConstructor;
+    private final org.jooq.Field<?>[] recordFields;
+    private final Map<String, Setter> setters;
+
+    public DomainRecordMapper(RecordType<R> recordType, Class<? extends E> type) {
+        try {
+            defaultConstructor = LOOKUP.findConstructor(type, MethodType.methodType(void.class));
+            recordFields = recordType.fields();
+            var fieldNames =
+                    Arrays.stream(recordFields).map(org.jooq.Field::getName).collect(Collectors.toSet());
+            setters = getSetters(fieldNames, type);
+        } catch (Exception e) {
+            throw new RecordMappingException(
+                    String.format("Failed to create record mapper for entity type %s", type.getName()), e);
+        }
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public @Nullable E map(R source) {
+        try {
+            var entity = (E) defaultConstructor.invoke();
+            for (int i = 0; i < recordFields.length; i++) {
+                var field = recordFields[i];
+                var setter = setters.get(field.getName());
+                if (setter == null) {
+                    continue;
```

Should this throw an error or are there valid scenarios where it can be missing?

---

good catch. This should never happen, so I'll remove it. I think I was considering the inverted scenario, that we do have some transient field / some setters without corresponding column in our domain entity class